libs/libarchfpga/src/read_fpga_interchange_arch.cpp

```diff
@@ -163,6 +234,220 @@ struct ArchReader {
             }
         }
     }
+
+    // Layout Processing
+    void process_layout() {
+        auto strList = ar_.getStrList();
+        auto tileList = ar_.getTileList();
+        auto tileTypeList = ar_.getTileTypeList();
+        t_grid_def grid_def;
+
+        grid_def.width = grid_def.height = 0;
+        for (auto tile : tileList) {
+            grid_def.width = std::max(grid_def.width, tile.getCol() + 1);
+            grid_def.height = std::max(grid_def.height, tile.getRow() + 1);
+        }
+
+        grid_def.grid_type = GridDefType::FIXED;
+        std::string name = std::string(ar_.getName());
+        if (name == "auto") {
+            archfpga_throw(arch_file_, __LINE__,
+                           "The name auto is reserved for auto-size layouts; please choose another name");
+        }
+        grid_def.name = name;
+        for (auto tile : tileList) {
+            t_metadata_dict data;
+            std::string tile_prefix(strList[tile.getName()].cStr());
+            auto tileType = tileTypeList[tile.getType()];
+            std::string tile_type(strList[tileType.getName()].cStr());
+
+            size_t pos = tile_prefix.find(tile_type);
+            if (pos != std::string::npos && pos == 0)
+                tile_prefix.erase(pos, tile_type.length() + 1);
+            data.add(arch_->strings.intern_string(vtr::string_view("fasm_prefix")),
+                     arch_->strings.intern_string(vtr::string_view(tile_prefix.c_str())));
+            t_grid_loc_def single(tile_type, 1);
+            single.x.start_expr = tile.getCol();
+            single.y.start_expr = tile.getRow();
+            single.x.end_expr = single.x.start_expr + " + w - 1";
+            single.y.end_expr = single.y.start_expr + " + h - 1";
+            single.owned_meta = std::make_unique<t_metadata_dict>(data);
+            single.meta = single.owned_meta.get();
+            grid_def.loc_defs.emplace_back(std::move(single));
+        }
+
+        arch_->grid_layouts.emplace_back(std::move(grid_def));
+    }
+
+    void process_device() {
+        /*
```

Comment:
Make sense. Should we add ``FIXME`` or ``TODO`` to highlight that there are some pending work to do?

Reply:
Added a `FIXME`